inst/tests/tests.Rraw

```diff
@@ -15194,6 +15194,78 @@ l = replicate(ceiling(log10(.Machine$integer.max)), rep(1L, 10L), simplify = FAL
+test(2062.021, fifelse(test_vec_na, rep(1L,12L), 0L), out_vec_na)
+test(2062.022, fifelse(test_vec_na, rep(1L,12L), rep(0L,12L)), out_vec_na)
+test(2062.023, fifelse(test_vec_na, 1L, rep(0L,12L)), out_vec_na)
+test(2062.024, fifelse(test_vec_na, 1, 0), as.numeric(out_vec_na))
+test(2062.025, fifelse(test_vec_na, rep(1,12L), 0), as.numeric(out_vec_na))
+test(2062.026, fifelse(test_vec_na, rep(1,12L), rep(0,12L)), as.numeric(out_vec_na))
+test(2062.027, fifelse(test_vec_na, 1, rep(0,12L)), as.numeric(out_vec_na))
+test(2062.028, fifelse(test_vec_na, TRUE, rep(FALSE,12L)), as.logical(out_vec_na))
+test(2062.029, fifelse(test_vec_na, rep(TRUE,12L), FALSE), as.logical(out_vec_na))
+test(2062.030, fifelse(test_vec_na, rep(TRUE,12L), rep(FALSE,12L)), as.logical(out_vec_na))
+test(2062.031, fifelse(test_vec_na, "1", rep("0",12L)), as.character(out_vec_na))
+test(2062.032, fifelse(test_vec_na, rep("1",12L), "0"), as.character(out_vec_na))
+test(2062.033, fifelse(test_vec_na, rep("1",12L), rep("0",12L)), as.character(out_vec_na))
+test(2062.034, fifelse(test_vec_na, "1", "0"), as.character(out_vec_na))
+test(2062.035, fifelse(test_vec, as.Date("2011-01-01"), FALSE), error = "Item 'yes' is of type double but item 'no' is of type logical. Please make sure they are of the same type.")
+test(2062.036, fifelse(test_vec_na, 1+0i, 0+0i), as.complex(out_vec_na))
+test(2062.037, fifelse(test_vec_na, rep(1+0i,12L), 0+0i), as.complex(out_vec_na))
+test(2062.038, fifelse(test_vec_na, rep(1+0i,12L), rep(0+0i,12L)), as.complex(out_vec_na))
+test(2062.039, fifelse(test_vec_na, 1+0i, rep(0+0i,12L)), as.complex(out_vec_na))
+test(2062.040, fifelse(test_vec, as.raw(0), as.raw(1)), error = "Type raw is not supported.")
+test(2062.041, fifelse(TRUE,1,as.Date("2019-07-07")), error = "Item 'yes' has different class than item 'no'. Please make sure that they have same class type.")
+test(2062.042, fifelse(TRUE,1L,factor(letters[1])), error = "Item 'yes' has different class than item 'no'. Please make sure that they have same class type.")
+test(2062.043, fifelse(TRUE, list(1:5), list(5:1)), list(1:5))
+test(2062.044, fifelse(as.logical(NA), list(1:5), list(5:1)), list(as.integer(NA)))
+test(2062.045, fifelse(FALSE, list(1:5), list(5:1)), list(5:1))
+test(2062.046, fifelse(TRUE, data.table(1:5), data.table(5:1)), data.table(1:5))
+test(2062.047, fifelse(FALSE, data.table(1:5), data.table(5:1)), data.table(5:1))
+test(2062.048, fifelse(TRUE, data.frame(1:5), data.frame(5:1)), data.frame(1:5))
+test(2062.049, fifelse(FALSE, data.frame(1:5), data.frame(5:1)), data.frame(5:1))
+test(2062.050, fifelse(c(TRUE,FALSE), list(1:5,6:10), list(10:6,5:1)), list(1:5,5:1))
+test(2062.051, fifelse(c(as.logical(NA),TRUE), list(1:5,6:10), list(10:6,5:1)), list(as.integer(NA),6:10))
+test(2062.052, fifelse(c(FALSE,TRUE), list(1:5,6:10), list(10:6,5:1)), list(10:6,6:10))
+test(2062.053, fifelse(c(as.logical(NA),TRUE), list(1:5), list(10:6,5:1)), list(as.integer(NA),1:5))
```

Comment:
This doesn't error because `yes` is treated as a singleton, right? I believe this is confirmed by test 55 below

Reply:
I added test 61 for the singleton